EIPS/eip-7254.md

```diff
@@ -0,0 +1,174 @@
+---
+eip: 7254
+title: Token Revenue Sharing
+description: Revenue token is a token that shares rewards for holders.
+author: Quy Phan (@quyphandang), Quy Phan <quy.phan@cryptoviet.info>
+discussions-to: https://ethereum-magicians.org/t/token-revenue-sharing/14872
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-06-29
+---
+
+## Abstract
+
+Revenue Sharing is an extended version of the standard token. It proposes an additional payment method for token holders. 
+
+This standard includes updating rewards for holders and allowing token holders to withdraw rewards.
+
+
+## Specification
+
+### Methods
+
+#### name
+
+Returns the name of the token
+
+``` js
+function name() public view returns (string)
+```
+
+
+#### symbol
+
+Returns the symbol of the token.
+
+``` js
+function symbol() public view returns (string)
+```
+
+
+
+#### decimals
+
+Returns the number of decimals the token uses
+
+``` js
+function decimals() public view returns (uint8)
+```
+
+
+#### totalSupply
+
+Returns the total token supply.
+
+``` js
+function totalSupply() public view returns (uint256)
+```
+
+
+
+#### balanceOf
+
+Returns the account balance of another account with address `account`.
+
+``` js
+function balanceOf(address account) public view returns (uint256)
+```
+
+
+
+#### informationOf
+
+Returns the account information of another account with the address `account`, including: inReward, outReward and withdraw.
+
+``` js
+function informationOf(address account) public view returns (UserInformation memory)
+```
+
+
+
+#### transfer
+
+Transfers `amount` amount of tokens to address `to`.
+
+``` js
+function transfer(address to, uint256 amount) public returns (bool)
+```
+
+
+
+#### transferFrom
+
+Transfers `amount` amount of tokens from address `from` to address `to`.
+
+``` js
+function transferFrom(address from, address to, uint256 amount) public returns (bool)
+```
+
+
+
+#### approve
+
+Allows `spender` to withdraw from your account multiple times, up to the `amount` amount.
+
+``` js
+function approve(address spender, uint256 amount) public returns (bool)
+```
+
+
+#### allowance
+
+Returns the amount which `spender` is still allowed to withdraw from `owner`.
+
+``` js
+function allowance(address owner, address spender) public view returns (uint256)
+```
+
+
+
+#### tokenReward
+
+Returns the token reward address of the token.
```

Comment:
"Reward" has not been defined in a normative section, nor have the mechanics of the "reward" been specified.